### PR TITLE
Feature/extensions

### LINKED ISSUE #4035: When reordering columns they revert back to original order
I was able to get the plugin to work. I can click on a column and drag it, but when I let go of the column it just jumps back to the original position.

What's weird is that even your example page here does the same thing. Am I missing something as to the functionality of this extension? It is my understanding that I should be able to reorder the columns.

http://issues.wenzhixin.net.cn/bootstrap-table/#extensions/reorder-columns.html

### PR DESCRIPTION
**Pull request by wenzhixin**
* Fixed CSS error
* Fixed cookie search cannot work bug
* Fixed toolbar display error

fix #4035

TODO:

- [x] bootstrap-table-reorder-columns cannot work.

### CODE REVIEW

File: src/extensions/cookie/bootstrap-table-cookie.js [hunk 1]
```diff
@@ -329,7 +329,7 @@ $.BootstrapTable = class extends $.BootstrapTable {
   onSearch (event) {
     super.onSearch(event)
 
-    if ($(event.currentTarget).parent().hasClass('search')) {
+    if (this.options.search) {
```
Comment (djhvscf) [inline src/extensions/cookie/bootstrap-table-cookie.js]: Is this change equivalent?

File: src/extensions/sticky-header/bootstrap-table-sticky-header.css [hunk 2]
```diff
@@ -17,6 +17,6 @@
   background: #e9ecef;
 }
 
-.fix-sticky table thead.thead-light {
+.fix-sticky table thead.thead-dark {
```
Comment (djhvscf) [inline src/extensions/sticky-header/bootstrap-table-sticky-header.css]: Why is this change? The hex color is the same

Comment (wenzhixin) [reply src/extensions/sticky-header/bootstrap-table-sticky-header.css]: The last version was wrong: https://github.com/wenzhixin/bootstrap-table/blob/develop/src/extensions/sticky-header/bootstrap-table-sticky-header.css#L16

File: src/extensions/cookie/bootstrap-table-cookie.js (hunk 1 above)
Comment (wenzhixin) [reply src/extensions/cookie/bootstrap-table-cookie.js]: Fix `$(event.currentTarget).parent().hasClass('search')` always return `false` bug.

**Comment by UtechtDustin** [comment]
I will review it as soon as WIP is removed.

**Comment by UtechtDustin** [comment]
> Fixed cookie search cannot work bug

This Bug #4158 ?

**Comment by UtechtDustin** [comment]
> bootstrap-table-reorder-columns cannot work.

Fixed, demo: https://live.bootstrap-table.com/code/UtechtDustin/35

**Comment by UtechtDustin** [comment]
fix #4035

**Comment by wenzhixin** [comment]
Fixed export dropdown error.
error: https://live.bootstrap-table.com/code/wenzhixin/44
fixed: https://live.bootstrap-table.com/code/wenzhixin/45 

image

**Comment by djhvscf** [comment]
> > bootstrap-table-reorder-columns cannot work.
> 
> Fixed, demo: https://live.bootstrap-table.com/code/UtechtDustin/35

This is fixed. Please test it under the same live example above

File: src/extensions/reorder-columns/bootstrap-table-reorder-columns.js [hunk 3]
```diff
@@ -135,16 +135,19 @@ BootstrapTable.prototype.makeRowsReorderable = function () {
...
         if (columnIndex !== -1) {
           that.columns[columnIndex].fieldIndex = i
           columns.push(that.columns[columnIndex])
-          that.columns.splice(columnIndex, 1)
         }
       }
 
-      that.columns = that.columns.concat(columns)
+      for (let i = 0; i < ths.length; i++ ) {
+        that.fieldsColumnsIndex[ths[i]] = i
```
Comment (UtechtDustin) [inline src/extensions/reorder-columns/bootstrap-table-reorder-columns.js]: Can be done in the for loop at line 138, no need for an new for loop.

**Pull request merged**